src/app/utils/field-validation/validators/EmailValidator.class.js

```diff
@@ -4,7 +4,22 @@ const validator = require('validator')
 class EmailValidator extends BaseFieldValidator {
   _isFilledAnswerValid() {
     const { answer } = this.response
-    const isValid = validator.isEmail(String(answer))
+    const {
+      isVerifiable,
+      hasAllowedEmailDomains,
+      allowedEmailDomains,
+    } = this.formField
+    const emailAddress = String(answer)
+    let isValid = validator.isEmail(emailAddress)
+    if (
+      isValid &&
+      isVerifiable &&
```

Comment:
I don't think we need the `isVerifiable` portion, that is only for verified fields, which has nothing to do with this feature

Reply:
Current implementation allows one to be checked and the other to be unchecked but the requirement is to enable validation of email domains only when both are checked.

I am supposed to make a change such that the validations toggles are coupled in this way:

1. If `isVerifiable` and `hasAllowedEmailDomains` are both `true`, and I switch off `isVerifiable`, then `hasAllowedEmailDomains` gets switched off automatically
1. If `isVerifiable` is `false`, then I cannot switch on `hasAllowedEmailDomains`

From just the above description, I am not sure if this is giving users the right user experience and whether users can easily understand the relationship between these two validation toggles.

Reply:
I guess you can hide the allow domain toggle when isVerifiable toggle is off, similar to how the email autoreply toggle is activated/hidden. Might want to clarify with the PMs!